docs/architecture/glossary.rst

```diff
@@ -0,0 +1,67 @@
+Glossary
```

Comment:
Does this come from the wiki? https://github.com/SymbiFlow/prjxray/wiki/Glossary ?

Reply:
Yes and no.  I pulled some definitions from there but tried to keep the focus on overall architecture so I didn't pull them all.